core/services/directrequestocr/drlistener.go

```diff
@@ -0,0 +1,271 @@
+	ParseErrorTaskName  string = "parse_error"
+)
+
+type DRListener struct {
+	oracle            ocr2dr_oracle.OCR2DROracle
+	job               job.Job
+	pipelineRunner    pipeline.Runner
+	jobORM            job.ORM
+	logBroadcaster    log.Broadcaster
+	runs              sync.Map
+	shutdownWaitGroup sync.WaitGroup
+	mbOracleEvents    *utils.Mailbox[log.Broadcast]
+	chStop            chan struct{}
+	pluginORM         ORM
+	pluginConfig      config.PluginConfig
+	logger            logger.Logger
+	utils.StartStopOnce
+}
+
+func NewDRListener(oracle ocr2dr_oracle.OCR2DROracle, jb job.Job, runner pipeline.Runner, jobORM job.ORM, pluginORM ORM, pluginConfig config.PluginConfig, logBroadcaster log.Broadcaster, lggr logger.Logger) *DRListener {
+	return &DRListener{
+		oracle:         oracle,
+		job:            jb,
+		pipelineRunner: runner,
+		jobORM:         jobORM,
+		logBroadcaster: logBroadcaster,
+		mbOracleEvents: utils.NewHighCapacityMailbox[log.Broadcast](),
+		chStop:         make(chan struct{}),
+		pluginORM:      pluginORM,
+		pluginConfig:   pluginConfig,
+		logger:         lggr,
+	}
+}
+
+// Start complies with job.Service
+func (l *DRListener) Start(context.Context) error {
+	return l.StartOnce("DirectRequestListener", func() error {
+		unsubscribeLogs := l.logBroadcaster.Register(l, log.ListenerOpts{
+			Contract: l.oracle.Address(),
+			ParseLog: l.oracle.ParseLog,
+			LogsWithTopics: map[common.Hash][][]log.Topic{
+				ocr2dr_oracle.OCR2DROracleOracleRequest{}.Topic():  {},
+				ocr2dr_oracle.OCR2DROracleOracleResponse{}.Topic(): {},
+			},
+			MinIncomingConfirmations: l.pluginConfig.MinIncomingConfirmations,
+		})
+		l.shutdownWaitGroup.Add(2)
+		go l.processOracleEvents()
+		go func() {
+			<-l.chStop
+			unsubscribeLogs()
+			l.shutdownWaitGroup.Done()
+		}()
+
+		return nil
+	})
+}
+
+// Close complies with job.Service
+func (l *DRListener) Close() error {
+	return l.StopOnce("DirectRequestListener", func() error {
+		l.runs.Range(func(key, runCloserChannelIf interface{}) bool {
+			runCloserChannel, _ := runCloserChannelIf.(chan struct{})
+			close(runCloserChannel)
+			return true
+		})
+		l.runs = sync.Map{}
+
+		close(l.chStop)
+		l.shutdownWaitGroup.Wait()
+
+		return nil
+	})
+}
+
+// HandleLog() complies with log.Listener
+func (l *DRListener) HandleLog(lb log.Broadcast) {
+	log := lb.DecodedLog()
+	if log == nil || reflect.ValueOf(log).IsNil() {
+		l.logger.Error("HandleLog: ignoring nil value")
+		return
+	}
+
+	switch log := log.(type) {
+	case *ocr2dr_oracle.OCR2DROracleOracleRequest, *ocr2dr_oracle.OCR2DROracleOracleResponse:
+		wasOverCapacity := l.mbOracleEvents.Deliver(lb)
+		if wasOverCapacity {
+			l.logger.Error("OracleRequest log mailbox is over capacity - dropped the oldest log")
+		}
+	default:
+		l.logger.Warnf("Unexpected log type %T", log)
+	}
+}
+
+// JobID() complies with log.Listener
+func (l *DRListener) JobID() int32 {
+	return l.job.ID
+}
+
+func (l *DRListener) processOracleEvents() {
```

Comment:
hmm I still think its much simpler to use the lp and query EA directly. E.g.
```
tick := time.After(0)
for {
     select {
          case <-l.chStop():
               // end
          case <-tick:
               tick = time.After(X seconds)
               latest, _  := c.lp.LatestBlock(pg.WithParentCtx(l.ctx)
               requests, _ := c.lp.LogsWithSigsExcluding(
		 latest-l.lookbackBlocks, // set to max time to response inclusion 
		 latest-l.minConfs,
		 common.Hash{
			l.ocr2dr_oracle.OCR2DROracleOracleRequest{}.Topic(),
		  },
		  common.Hash{
			l.ocr2dr_oracle.OCR2DROracleOracleResponse{}.Topic()},
		  1, // Match on word 1 (requestID)
		  l.address,
		  pg.WithParentCtx(ctx))
		for _, req := range requests {
		   if _, ok := l.resultsReadyCache[req.ID]; ok {
			continue
		   }
		   result, err := queryEA(l.ctx, ...)
		   l.resultsReadyCache[req.ID] = Result{result, err}
		 }
     }
```
then just expose l.resultsReadyCache to OCR. To handle node restart you could either persist the cache or just let the EA handle duplicate reqs (maybe it already does?).